Src/IdiomsUnitTest/GuardClauseAssertionTest.cs

```diff
@@ -1727,6 +1728,11 @@ public void Method(string argument1, out object argument2, int argument3)
                     throw new ArgumentNullException(nameof(argument1));
                 }
 
+                if (argument1 == string.Empty)
```

Comment:
I believe you are correct; once I remove my breaking change, I won't need the change I made here.

Reply:
Do we still need the integration tests if I removed the new empty string guard from default behavior?

Reply:
Yes, we definitely need.

Currently we have two kind of tests: unit and integration (or facade). Previously you have added a set of unit tests in the `EmptyStringBehaviorExpectationTest` file to verify that it behaves correctly. However, it would be cool to also have tests where we initialize real guard assertion with our custom behavior and verify the real methods. While tests overlap a bit, I believe that we still have a lot of benefits from it.